src/Paket.Core/embedded/Paket.Restore.targets

```diff
@@ -75,14 +67,38 @@
     <PaketIntermediateOutputPath Condition=" '$(PaketIntermediateOutputPath)' == '' ">$(BaseIntermediateOutputPath.TrimEnd('\').TrimEnd('\/'))</PaketIntermediateOutputPath>
   </PropertyGroup>
 
+  <!-- Check if paket is available as local dotnet cli tool -->
+  <Target Name="SetPaketCommand" >
+    <!-- Call 'dotnet paket' and see if it returns without an error. Mute all the output. -->
+    <Exec Command="dotnet paket --version" IgnoreExitCode="true" StandardOutputImportance="low" StandardErrorImportance="low" >
+      <Output TaskParameter="ExitCode" PropertyName="LocalPaketToolExitCode" />
+    </Exec>
+
+    <!-- Then we go through our normal steps of setting the Paket command.
+         I called it InternalPaketCommand here because apparently properties aren't available
+         outside the targets they are declared in. -->
+    <PropertyGroup>
+      <_PaketExeExtension>$([System.IO.Path]::GetExtension("$(PaketExePath)"))</_PaketExeExtension>
+      <InternalPaketCommand Condition=" '$(InternalPaketCommand)' == '' AND '$(_PaketExeExtension)' == '.dll' ">dotnet "$(PaketExePath)"</InternalPaketCommand>
+      <InternalPaketCommand Condition=" '$(InternalPaketCommand)' == '' AND '$(_PaketExeExtension)' == '' AND $(LocalPaketToolExitCode) == 0">dotnet "$(PaketExePath)"</InternalPaketCommand>
```

Comment:
I guess we could start with checking if `LocalPaketToolExitCode == 0` and if it is, just set `InternalPaketCommand` to `dotnet paket`? @NatElkins

Reply:
Sure whatever you think is clearest to someone coming back a year from now to try and decipher this.